### PR TITLE
CTD when firing a less-than-35-sprite bullet

### PR DESCRIPTION
**Pull request by GixG**

### CODE REVIEW

File: src/Battlescape/Map.cpp [hunk 1]
```diff
@@ -937,9 +937,10 @@ void Map::drawTerrain(Surface *surface)
 
 								for (int i = begin; i != end; i += direction)
 								{
-									tmpSurface = _projectileSet->getFrame(_projectile->getParticle(i));
-									if (tmpSurface)
+									Surface *ts = _projectileSet->getFrame(_projectile->getParticle(i));
```
Comment (Yankes) [inline src/Battlescape/Map.cpp]: could you fix real problem not symptoms? This two lines are correct, where is error?

Comment (GixG) [reply src/Battlescape/Map.cpp]: Well, the getFrame returns  (Surface *)nullptr when no sprite defined. But the issue is the automatic dereference of nullptr awhen assigning a tmpSurface which is SurfaceRaw<const Uint8>... You mean then maybe change the return type of getFrame?

Comment (Yankes) [reply src/Battlescape/Map.cpp]: Ok, right, error is on my part, small design hole, `SurfaceRaw` should accept `nullptr` without crash. And this is true error. Not this one line.

Comment (Yankes) [reply src/Battlescape/Map.cpp]: I think I will fix it myself, thanks for spotting this bug.

**Pull request closed without merging**